### PR TITLE
Pro users: Add a test

### PR DESCRIPTION
**Pull request by mlahargou**
This verifies that pro users get a discount on products. We are using
the repair business toolkit and mocking the logged in user to verify
that they get the discount.

Because we are not mocking the product itself, I didn't want to get
too specific with prices. Instead, we just verify that the price
for a regular user is *higher* than it is for a pro user.

Connects: #1043 

cc @batbattur

### CODE REVIEW

**Review by batbattur — APPROVED** [review]
CR 🚛

File: frontend/test/cypress/integration/product/pro_user.cy.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+describe('Pro user test', () => {
+   it('will give pro users a discount', () => {
+      cy.loadProductPageByPath('/products/repair-business-toolkit');
+
+      // Get price from page
+      cy.findAllByTestId('product-price')
+         .first()
```
Comment (ardelato) [inline frontend/test/cypress/integration/product/pro_user.cy.ts:7]: Is there a possibility of the price for a recommended product being added to the list first? In the case, the actual product price does not render or is slower at rendering?

**Review by ardelato — APPROVED** [review]
CR 👍 with question

**Comment by jordycosta** [comment]
QA 🟢 

https://github.com/iFixit/react-commerce/actions/runs/3500010244/jobs/5862219029

test

**Pull request merged**